Makefile

```diff
@@ -702,51 +708,52 @@ localmachine:
+
+export \
+	REMOTESYSTEM_TCP_PORT \
+	REMOTESYSTEM_TLS_CA_KEY \
+	REMOTESYSTEM_TLS_CA_CRT \
+	REMOTESYSTEM_TLS_SERVER_CRT \
+	REMOTESYSTEM_TLS_SERVER_KEY \
+	REMOTESYSTEM_TLS_CLIENT_CRT \
+	REMOTESYSTEM_TLS_CLIENT_KEY \
+	REMOTESYSTEM_TLS_BOGUS_CRT \
+	REMOTESYSTEM_TLS_BOGUS_KEY
+
+$(REMOTESYSTEM_TLS_CA_CRT) $(REMOTESYSTEM_TLS_CA_KEY) \
+$(REMOTESYSTEM_TLS_CLIENT_CRT) $(REMOTESYSTEM_TLS_CLIENT_KEY) \
+$(REMOTESYSTEM_TLS_SERVER_CRT) $(REMOTESYSTEM_TLS_SERVER_KEY) \
+$(REMOTESYSTEM_TLS_BOGUS_CRT) $(REMOTESYSTEM_TLS_BOGUS_KEY) \
+&:
+	source hack/remotesystem.env ; remotesystem-gen-tls
+
 .PHONY: localsystem
-localsystem:
+localsystem: $(REMOTESYSTEM_TLS_CA_CRT) $(REMOTESYSTEM_TLS_SERVER_CRT) $(REMOTESYSTEM_TLS_CLIENT_CRT) $(REMOTESYSTEM_TLS_BOGUS_CRT)
 	# Wipe existing config, database, and cache: start with clean slate.
 	$(RM) -rf ${HOME}/.local/share/containers ${HOME}/.config/containers
 	PODMAN=$(CURDIR)/bin/podman QUADLET=$(CURDIR)/bin/quadlet bats -T --filter-tags '!ci:parallel' test/system/
 	PODMAN=$(CURDIR)/bin/podman QUADLET=$(CURDIR)/bin/quadlet bats -T --filter-tags ci:parallel -j $$(nproc) test/system/
 
+
 .PHONY: remotesystem
-remotesystem:
+remotesystem: $(REMOTESYSTEM_TLS_CA_CRT) $(REMOTESYSTEM_TLS_SERVER_CRT) $(REMOTESYSTEM_TLS_CLIENT_CRT) $(REMOTESYSTEM_TLS_BOGUS_CRT)
 	# Wipe existing config, database, and cache: start with clean slate.
 	$(RM) -rf ${HOME}/.local/share/containers ${HOME}/.config/containers
-	# . Make sure there's no active podman server - if there is,
-	#   it's not us, and we have no way to know what it is.
-	# . Start server. Wait to make sure it comes up.
-	# . Run tests, pretty much the same as localsystem.
-	# . Stop server.
-	rc=0;\
-	if timeout -v 1 true; then \
-		if ./bin/podman-remote info; then \
-			echo "Error: podman system service (not ours) is already running" >&2;\
-			exit 1;\
-		fi;\
-		./bin/podman system service --timeout=0 > $(if $(PODMAN_SERVER_LOG),$(PODMAN_SERVER_LOG),/dev/null) 2>&1 & \
-		retry=5;\
-		while [ $$retry -ge 0 ]; do\
-			echo Waiting for server...;\
-			sleep 1;\
-			./bin/podman-remote info >/dev/null 2>&1 && break;\
-			retry=$$(expr $$retry - 1);\
-		done;\
-		if [ $$retry -lt 0 ]; then\
-			echo "Error: ./bin/podman system service did not come up" >&2;\
-			exit 1;\
-		fi;\
-		env PODMAN="$(CURDIR)/bin/podman-remote" bats -T --filter-tags '!ci:parallel' test/system/ ;\
-		rc=$$?; \
-		if [ $$rc -eq 0 ]; then \
-		   env PODMAN="$(CURDIR)/bin/podman-remote" bats -T --filter-tags ci:parallel -j $$(nproc) test/system/ ;\
-		   rc=$$?;\
-		fi; \
-		kill %1;\
-	else \
-		echo "Skipping $@: 'timeout -v' unavailable'";\
-	fi;\
-	exit $$rc
+	source hack/remotesystem.env ; \
+	set -x ; \
+	remotesystem-ensure-timeout-cmd $@ || exit ; \
+	remotesystem-podman-service $(REMOTESYSTEM_TRANSPORT) $(PODMAN_SERVER_LOG) && \
+	remotesystem-wait-podman-service $(REMOTESYSTEM_TRANSPORT) && \
+	remotesystem-bats $(REMOTESYSTEM_TRANSPORT) $(CURDIR)/bin/podman-remote
```

Comment:
I would just have the script do everything directly instead of spiting this awkwardly with source here into all these functions.

Also I think it is safe to drop the `timeout` check from it. It really servers no purpose to skip the tests, we can just let them fail hard in that case.

Reply:
I intended to reproduce the embedded script as it was, but if that part is no longer necessary, I will remove it. I will bundle these all into a single "remotesystem-run" function.